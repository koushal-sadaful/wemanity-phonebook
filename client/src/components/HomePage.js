import React from 'react'
import {push} from 'connected-react-router'
import {bindActionCreators} from 'redux'
import {connect} from 'react-redux'
import {Container, Header, Search, Grid, Select, Input} from "semantic-ui-react";
import _ from 'lodash'
import {fetchAllEntries, resetState, updateSearchInputValue} from "../actions/HomePageActions";


class HomePage extends React.Component {

    searchInputUpdated = (e, {value}) => {
        console.log(value)
        this.props.updateSearchInputValue(value);
    };

    componentWillUnmount() {
        this.props.resetState()
    }

    render() {
        return (
            <div>
                <Container fluid>
                    <Container textAlign='center'>
                        <Header
                            as='h1'
                            content="Wemanity PhoneBook"
                            style={{
                                fontSize: '1.2em',
                                marginTop: '2em',
                            }}
                        />
                        <Header
                            as='h5'
                            style={{
                                fontSize: '1.2em',
                                fontWeight: 'normal',
                                marginBottom: 0,
                            }}
                        >

                            <Input
                                size={"large"}
                                icon={{name: 'search', circular: true, link: true}}
                                placeholder='Search phone number...'
                                onChange={_.debounce(this.searchInputUpdated, 500, {leading: true})}
                            />


                        </Header>
                    </Container>



                </Container>
            </div>
        )
    }
}


const mapStateToProps = ({homePage}) => ({
    allEntries: homePage.allEntries,
    errorMessage: homePage.errorMessage,
    searchValue: homePage.searchValue
});

const mapDispatchToProps = dispatch => bindActionCreators({
    updateSearchInputValue,
    fetchAllEntries,
    resetState,
}, dispatch);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(HomePage)