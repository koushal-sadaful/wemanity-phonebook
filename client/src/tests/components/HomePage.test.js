import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, mount} from 'enzyme';
import {Header, Input, Search} from "semantic-ui-react";
import HomePage from "../../components/HomePage";
import {createTestStore} from "../testStoreFactory";
import axios from "axios";
import {runTheSnapshotTest} from "../testUtils/TestHelpers";

configure({adapter: new Adapter()});
jest.mock('axios');
jest.useFakeTimers()

describe("HomePage", () => {
    let wrapper = null;
    afterEach(() => {
        console.log(wrapper)
        if (wrapper){
            wrapper.unmount()
            wrapper = null
        }

    });

    it("should render a clean search page", () => {
        wrapper = mount(<HomePage store={createTestStore()}/>);

        const pageTitle = wrapper.find(Header).first().props().content;
        expect(pageTitle).toMatch("Wemanity PhoneBook");

        const searchBarProps = wrapper.find(Input).props();
        expect(searchBarProps.placeholder).toMatch("Search phone number...");

    });

    it("should pass the snapshot test", () => {
        runTheSnapshotTest(<HomePage store={createTestStore()}/>)
    });
});