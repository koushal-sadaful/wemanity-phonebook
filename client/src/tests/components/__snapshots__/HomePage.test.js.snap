// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should pass the snapshot test 1`] = `
<div>
  <div
    className="ui fluid container"
  >
    <div
      className="ui center aligned container"
    >
      <h1
        className="ui header"
        style={
          Object {
            "fontSize": "1.2em",
            "marginTop": "2em",
          }
        }
      >
        Wemanity PhoneBook
      </h1>
      <h5
        className="ui header"
        style={
          Object {
            "fontSize": "1.2em",
            "fontWeight": "normal",
            "marginBottom": 0,
          }
        }
      >
        <div
          className="ui large icon input"
        >
          <input
            onChange={[Function]}
            placeholder="Search phone number..."
            type="text"
          />
          <i
            aria-hidden="true"
            className="search circular link icon"
            onClick={[Function]}
          />
        </div>
      </h5>
      <div
        className="all-entreies"
      >
        <div
          className="ui three cards"
        >
          <div
            className="ui card home-page-entry"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                Sax
                , 
                John
              </div>
              <div
                className="description"
              >
                 
                +852 01 1257854
                  
              </div>
            </div>
            <div
              className="extra content"
            >
              <button
                className="ui blue basic button"
                onClick={[Function]}
              >
                 Edit 
              </button>
            </div>
          </div>
          <div
            className="ui card home-page-entry"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                Doe
                , 
                Jane
              </div>
              <div
                className="description"
              >
                 
                +852 112 155874
                  
              </div>
            </div>
            <div
              className="extra content"
            >
              <button
                className="ui blue basic button"
                onClick={[Function]}
              >
                 Edit 
              </button>
            </div>
          </div>
          <div
            className="ui card home-page-entry"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                Smith
                , 
                Will
              </div>
              <div
                className="description"
              >
                 
                +52 10 12548
                  
              </div>
            </div>
            <div
              className="extra content"
            >
              <button
                className="ui blue basic button"
                onClick={[Function]}
              >
                 Edit 
              </button>
            </div>
          </div>
          <div
            className="ui card home-page-entry"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                Jacobs
                , 
                Marc
              </div>
              <div
                className="description"
              >
                 
                +114 84 1478546
                  
              </div>
            </div>
            <div
              className="extra content"
            >
              <button
                className="ui blue basic button"
                onClick={[Function]}
              >
                 Edit 
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
